lib/simplecov_linter_formatter/summary_builder.rb

```diff
@@ -0,0 +1,127 @@
+module SimpleCovLinterFormatter
+  class SummaryBuilder
+    PERCENT_TEXT_SIZE = 3
+
+    def initialize(lines)
+      @lines = lines
+      @largest_file_name = 0
+      @largest_percentage = 0
+    end
+
+    def build
+      body = build_body
+      return body if body == ""
+
+      "#{build_title}\n#{body}#{divider}"
+    end
+
+    private
+
+    def build_title
+      "SimpleCov Report - (#{build_coverage_title})\n#{divider}"
+    end
+
+    def divider
+      '-' * report_width
+    end
+
+    def report_width
+      @largest_file_name + @largest_percentage + PERCENT_TEXT_SIZE
+    end
+
+    def build_coverage_title
+      if SimpleCovLinterFormatter.cover_all?
+        return "Total Coverage"
+      end
+
+      "Own Changes"
+    end
+
+    def build_body
+      result = ""
+
+      files.each do |file|
+        result += "#{build_line_output(file)}\n"
+      end
+
+      result
+    end
+
+    def files
+      unique_files = []
+      added_files = {}
+
+      @lines.each do |line|
+        file = SimpleCovLinterFormatter::FileLine.new(line)
+        next if added_files[file.file_short_name]
+
+        unique_files << file
+        added_files[file.file_short_name] = true
+        file_size = file.file_short_name.size
+        file_percentage = file.file_int_percentage.to_s.size
+        @largest_file_name = file_size if file_size > @largest_file_name
+        @largest_percentage = file_percentage if file_percentage > @largest_percentage
+      end
+
+      sort_files(unique_files)
+    end
+
+    def sort_files(files)
+      sorting_strategy = if SimpleCovLinterFormatter.summary_coverage_sorting?
+                           Proc.new { |file| [file.percentage, file.file_short_name] }
+                         else
+                           Proc.new { |file| file.file_short_name }
+                         end
+      files.sort_by { |file| sorting_strategy.call(file) }
+    end
+
+    def build_line_output(file)
+      file_name = adjusted_file_name(file)
+      covered_size = @largest_file_name * file.percentage / 100.0
+      covered_text = colorize_text(
+        covered_size.zero? ? "" : file_name[0..(covered_size - 1)],
+        :summary_covered_bg_color
+      )
+      not_covered_text = colorize_text(
+        file_name[covered_size..-1],
```

Comment:
**[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Style/SlicingWithRange: Prefer ary[n..] over ary[n..-1].

Reply:
does not work with ruby 2.5